File: Server/Hero.Server.Core/Models/Skill.cs
namespace Hero.Server.Core.Models
{
    public class Skill : Suggestable
    {
        public Guid? AbilityId { get; set; }
        public string? IconUrl { get; set; }
        public required string Name { get; set; }
        public string? Description { get; set; }
        public required List<AttributeSkill> Attributes { get; set; }
        public Ability? Ability { get; set; }

        public void Update(Skill skill)
        {
            this.AbilityId = skill.AbilityId;
            this.IconUrl= skill.IconUrl;
            this.Name= skill.Name;
            this.Description= skill.Description;   
        }

    }
}

File: Client/lib/src/features/traits/domain/skill.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:kalinar/src/features/traits/domain/suggestable.dart';

import '../../group_management/domain/user.dart';
import 'ability.dart';
import 'attribute_value.dart';
import 'suggestion_state.dart';

part 'skill.freezed.dart';
part 'skill.g.dart';

@freezed
class Skill with _$Skill implements Suggestable {
  const factory Skill(
    String id, {
    Ability? ability,
    String? iconUrl,
    required String name,
    String? description,
    @Default([]) List<AttributeValue> attributes,
    required User creator,
    required SuggestionState state,
    String? rejectionReason,
    required DateTime createdAt,
    DateTime? lastUpdatedAt,
    DateTime? approvedAt,
    DateTime? rejectedAt,
  }) = _Skill;

  factory Skill.fromJson(Map<String, dynamic> json) => _$SkillFromJson(json);
}
